feat: updated miscellaneous documentation (#51)
godot-escoria/escoria-docs: getting_started/dialogs.rst

    See :doc:`the ESC reference </scripting/z_esc_reference#conditions>` for
    details about conditions.

Recorded Speech
---------------

Escoria features voice support as well.

To use it, every line that should support a voice file requires an
additional text key (a name, followed by a colon, followed by the text to say),

> like this:

Reviewer: ```suggestion
additional text key (i.e. a name, followed by a colon, followed by the text to
say), like this:
```

.. code-block::

    say player worker_hello:"Hello!"

> The name of the audio file will act as a key for the `say` command so it

Reviewer: ```suggestion
The name of the audio file serves as a key for the `say` command so it
```

knows which audio file to play. The name of the file (without any extension)
must be the same as the key. As an example, the above `say` command would play
the audio file "worker_hello.mp3" (or any other supported audio format file

> like "worker_hello.ogg".

Reviewer: ```suggestion
like "worker_hello.ogg").
```

The audio formats that Godot supports are listed here :
:doc:`https://docs.godotengine.org/en/stable/tutorials/assets_pipeline/importing_audio_samples.html?highlight=ogg#supported-files`

> Escoria uses a configuration parameter to specify where in your file

Reviewer: ```suggestion
Escoria uses a configuration parameter to specify where in your directory
```

structure to find your game's audio files. This setting can be found in
`Project/Project Settings/Escoria/Sound/Speech Folder`. Set this to a
location appropriate for your game - e.g. `res://game/speech`.

See :doc:`https://docs.escoria-framework.org/en/devel/scripting/z_esc_reference.html#say-player-text-type-api-doc`

> for further details on the say command.

Reviewer: ```suggestion
for further details on the `say` command.
```

Translations
------------

The detail below is only a high-level overview of Internationalization support
in Godot. For more information, please see Godot's translation documentation
https://docs.godotengine.org/en/stable/tutorials/i18n/internationalizing_games.html

Creating text translations
~~~~~~~~~~~~~~~~~~~~~~~~~~

Escoria takes advantage of Godot's built-in translation functionality for
providing language support. Translation information is found in Godot's
`Project/Project Settings/Localization` menu (text in `Translations`, audio in
`Remaps`).

Text translation relies on CSV files, an example of which is::
  keys,en,es
  ROOM1_greeting,"Hello, friend!","Hola, amigo!"

Once the CSV file containing the translation text has been created, use
Godot's importer to import it (under

> Project/Project Settings/Localization/Tranlations/Add).

Reviewer: ```suggestion
`Project/Project Settings/Localization/Tranlations/Add`).
```

For further details on creating and importing translations see
https://docs.godotengine.org/en/stable/tutorials/assets_pipeline/importing_translations.html

Using text translations in your game
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The key ("ROOM1_greeting" in the above example) is used in the "say" script

Reviewer: ```suggestion
The key ("ROOM1_greeting" in the above example) is used in the `say` script
```

command to tell Escoria which translation to look for. Place this key with
a colon prior to the text in your script file::

  :look
  say player ROOM1_greeting:"Hello, friend!"

Creating audio translations
~~~~~~~~~~~~~~~~~~~~~~~~~~~

Create your audio files to match the ones in the game's original language.
Store these files in the same location as your original recordings.

While the files can be called whatever you like, keeping the same name as the
original file and adding a language identifier is an easy way to keep track of
your files. e.g. A file called `hello.ogg` might have matching files called

> `hello_de.ogg` for the German translation, and `hello_fr.ogg` for the french.

Reviewer: ```suggestion
`hello_de.ogg` for the German translation, and `hello_fr.ogg` for the French.
```

Using audio translations in your game
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The following is a high-level overview of the language remapping functionality
provided by Godot. For more in-depth documentation, please see
https://docs.godotengine.org/en/stable/tutorials/i18n/internationalizing_games.html?highlight=remaps#localizing-resources

Godot provides a mechanism to map files between the different languages you
provide for your game. The mapping function can be found under
Project/Project Settings/Localization/Remaps.

Use the `Add` button in the `Resources` part of the window, choosing the audio
file you wish to provide a translation for (e.g. `hello.ogg`). Once you've
added the file, highlight it, and use the `Add` button in the `Remaps by
Locale` section of the window. In the file browser that appears, find the
matching audio file in the new language (e.g. `hello_fr.ogg`). Next to this
file, use the `Locale` pulldown menu to tell Godot what language that file
features. Add more remaps if you are supporting additional languages.

> Repeat this process for every source file, and every translated

Reviewer: ```suggestion
Repeat this process for every source file and every translated
```
